kodilla-patterns/src/main/java/com/kodilla/patterns/builder/bigmac/Bigmac.java

```diff
@@ -19,89 +19,26 @@
         private String sauce;
         private List<String> ingredients = new ArrayList<>();
 
-        public BigmacBuldier bun(Components components) {
-            switch (components) {
-                case BUN_WITH_SESAME:
-                    this.bun = Components.BUN_WITH_SESAME.getComponent();
-                    break;
-
-                case BUN_WITHOUT_SESAME:
-                    this.bun = Components.BUN_WITHOUT_SESAME.getComponent();
-                    break;
-
-                default:
-                    this.bun = Components.BUN_WITHOUT_SESAME.getComponent();
-                    break;
-            }
-            return this;
+        public BigmacBuldier bun(BunType bunType) {
+           this.bun = bunType.getBunType();
```

Comment:
niepotrzebnie tu zamieniasz na String tego enuma (w pozostalych analogicznych miejscach podobnie). Co przeszkadza, aby this.bun bylo typu BunType? po to wprowadzamy enuma, aby go miec i skorzystac z tego, co oferuje :)